support/cas-server-support-consent-webflow/src/main/java/org/apereo/cas/web/consent/config/CasConsentReviewConfiguration.java

```diff
@@ -0,0 +1,138 @@
+    private ServicesManager servicesManager;
+
+    @Autowired
+    @Qualifier("webApplicationServiceFactory")
+    private ServiceFactory<WebApplicationService> webApplicationServiceFactory;
+
+    @Autowired
+    @Qualifier("consentRepository")
+    private ConsentRepository consentRepository;
+
+    @Autowired
+    @Qualifier("consentEngine")
+    private ConsentEngine consentEngine;
+
+    @Bean
+    public CasConsentReviewController casConsentReviewController() {
+        return new CasConsentReviewController(consentRepository, consentEngine);
+    }
+
+    @Bean
+    @RefreshScope
+    public CasConsentReviewSecurityInterceptor casConsentReviewSecurityInterceptor() {
+        final String authorizer = IsAuthenticatedAuthorizer.class.getSimpleName();
+        if (!config.getAuthorizers().containsKey(authorizer)) {
+            config.addAuthorizer(authorizer, new IsAuthenticatedAuthorizer());
+        }
+        return new CasConsentReviewSecurityInterceptor(config, CAS_CLIENT_NAME,
+                "securityHeaders,csrfToken,".concat(IsAuthenticatedAuthorizer.class.getSimpleName()));
+    }
+
+    /**
+    * Initialize consent service.
+    */
+    @PostConstruct
+    protected void registerConsentService() {
+        final Service callbackService = this.webApplicationServiceFactory.createService(
+                casProperties.getServer().getPrefix().concat("/consent.*"));
+        if (!this.servicesManager.matchesExistingService(callbackService)) {
+            LOGGER.debug("Initializing consent service [{}]", callbackService);
+
+            final RegexRegisteredService service = new RegexRegisteredService();
+            service.setEvaluationOrder(0);
+            service.setName(service.getClass().getSimpleName());
+            service.setDescription("CAS Consent Overview");
+            service.setServiceId(callbackService.getId());
+
+            LOGGER.debug("Saving consent service [{}] into the registry", service);
+            this.servicesManager.save(service);
+            this.servicesManager.load();
+        }
+    }
+
+    @Override
+    public void addInterceptors(final InterceptorRegistry registry) {
+        registry.addInterceptor(casConsentReviewSecurityInterceptor()).addPathPatterns("/consent")
+                .addPathPatterns("/consent/*");
+    }
+
+    /**
+    * The security interceptor for consent overview.
+    */
+    public static class CasConsentReviewSecurityInterceptor extends CasSecurityInterceptor implements HandlerInterceptor {
+
+        public CasConsentReviewSecurityInterceptor(final Config config, final String clients,
+                final String authorizers) {
+            super(config, clients, authorizers);
+        }
+
+        @Override
+        public void postHandle(final HttpServletRequest request, final HttpServletResponse response,
+                               final Object handler, final ModelAndView modelAndView) throws Exception {
+            if (modelAndView != null
```

Comment:
Could this be moved to the security interceptor? In which case you no longer need this class at all?